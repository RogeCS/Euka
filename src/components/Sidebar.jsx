import React from "react";
import Logo from "./Logo.jsx";
import { SidebarData } from "../code/SidebarData";

import "../styles/components/Sidebar.scss";

const Sidebar = () => {
  return (
    <div className="dashboard-sidebar">
      <div className="dashboard-sidebar__logo">
        <Logo color="primary" />
      </div>
      <ul className="dashboard-sidebar__list">
        {SidebarData.map((val, key) => (
          <li
            key={key}
            className={`dashboard-sidebar__row ${
              window.location.pathname == val.link
                ? "dashboard-sidebar__row--active"
                : ""
            }`}
            onClick={() => {
              window.location.pathname = val.link;
            }}
          >
            <div className="dashboard-sidebar__icon">{val.icon}</div>
            <div
              className={`dashboard-sidebar__title ${
                window.location.pathname == val.link
                  ? "dashboard-sidebar__title--active"
                  : ""
              }`}
            >
              {val.title}
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Sidebar;
