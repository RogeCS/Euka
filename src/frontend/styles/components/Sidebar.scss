@import "../global/variables.scss";

.dashboard-sidebar {
  height: 100vh;
  min-width: $sidebar-width;
  width: $sidebar-width;
  background-color: $white;
  position: fixed;
  z-index: 999;
}

.dashboard-sidebar--open {
  transition: 200ms;
  left: 0;
  user-select: none;
}

.dashboard-sidebar--close {
  left: -$sidebar-width;
  transition: 200ms;
}

.dashboard-sidebar__toggle {
  align-items: center;
  color: $white;
  background-color: $dark-gray;
  border-radius: 0 10px 10px 0;
  display: flex;
  font-size: 1rem;
  height: $header-height;
  justify-content: center;
  padding-right: 3px;
  position: absolute;
  left: 100%;
  transition: 0.3s ease-in-out;
  width: 1.5rem;

  &:hover {
    background-color: $dark-color;
    transition: 0.3s ease-in-out;
  }
}

.dashboard-sidebar--open .dashboard-sidebar__open {
  display: none;
}

.dashboard-sidebar--close .dashboard-sidebar__close {
  display: none;
}

.dashboard-sidebar__logo {
  align-items: center;
  display: flex;
  height: calc(2 * #{$header-height});
  justify-content: center;
}

.dashboard-sidebar__list {
  height: auto;
  padding: 0;
  width: 100%;
}

.dashboard-sidebar__link {
  text-decoration: none;
}

.dashboard-sidebar__row {
  align-items: center;
  color: $dark-color;
  display: flex;
  flex-direction: row;
  font-size: 1rem;
  height: $header-height;
  justify-content: center;
  list-style-type: none;
  margin: 0;
  transition: 0.1s ease-in-out;
  width: 100%;

  &:hover {
    background-color: $gray;
    cursor: pointer;
    transition: 0.1s ease-in-out;
  }
}

.dashboard-sidebar__row--active {
  color: $secondary;
}

.dashboard-sidebar__icon {
  display: grid;
  flex: 40%;
  place-items: center;
}

.dashboard-sidebar__title {
  padding: 5px 0;
  flex: 60%;
}

.dashboard-sidebar__title--active {
  border-right: 3px solid $dark-color;
}
