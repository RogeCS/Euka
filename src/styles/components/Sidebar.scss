@import "../global/variables.scss";

.dashboard-sidebar {
  height: 100%;
  width: 250px;
  background: $color;
}
