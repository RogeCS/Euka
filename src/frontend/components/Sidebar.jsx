import React from "react";
import { Link } from "react-router-dom";
import Logo from "./Logo.jsx";
import * as AiIcons from "react-icons/ai";
import { SidebarData } from "../code/SidebarData";

import "../styles/components/Sidebar.scss";

const Sidebar = ({ sidebar, toggleSidebar }) => {
  const [windowLocation, setWindowLocation] = React.useState(0);

  React.useEffect(() => {
    setWindowLocation(window.location.pathname);
  });

  return (
    <div
      className={
        sidebar
          ? "dashboard-sidebar dashboard-sidebar--open"
          : "dashboard-sidebar dashboard-sidebar--close"
      }
    >
      <div className="dashboard-sidebar__toggle" onClick={toggleSidebar}>
        <span className="dashboard-sidebar__close">
          <AiIcons.AiFillCaretLeft />
        </span>
        <span className="dashboard-sidebar__open">
          <AiIcons.AiFillCaretRight />
        </span>
      </div>
      <span className="dashboard-sidebar__logo">
        <Logo color="primary" />
      </span>
      <ul className="dashboard-sidebar__list">
        {SidebarData.map((val, key) => (
          <Link className="dashboard-sidebar__link" key={key} to={val.link}>
            <li
              key={key}
              className={`dashboard-sidebar__row ${
                windowLocation == val.link
                  ? "dashboard-sidebar__row--active"
                  : ""
              }`}
            >
              <span className="dashboard-sidebar__icon">{val.icon}</span>
              <div
                className={`dashboard-sidebar__title ${
                  windowLocation == val.link
                    ? "dashboard-sidebar__title--active"
                    : ""
                }`}
              >
                {val.title}
              </div>
            </li>
          </Link>
        ))}
      </ul>
    </div>
  );
};

export default Sidebar;
