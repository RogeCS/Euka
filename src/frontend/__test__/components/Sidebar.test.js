import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import Sidebar from "../../components/Sidebar.jsx";

describe("<Sidebar />", () => {
  const sidebar = mount(<Sidebar />);

  //Checamos si existe sidebar
  test("Render Sidebar component", () => {
    expect(sidebar.length).toEqual(1);
  });

  test("Sidebar has 1 list", () => {
    expect(sidebar.find("ul")).toHaveLength(1);
  });

  /**
  Los snapshots nos ayudan a guardar la estructura
  actual de un componente y nos permite comparar los
  cambios que hagamos en el mismo y decidir si aceptamos
  los cambios o no.
  */
  test("Sidebar Snapshot", () => {
    const sidebar = create(<Sidebar />);
    expect(sidebar.toJSON()).toMatchSnapshot();
  });
});
