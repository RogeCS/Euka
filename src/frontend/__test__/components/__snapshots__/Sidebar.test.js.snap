// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Sidebar Snapshot 1`] = `
<div
  className="dashboard-sidebar dashboard-sidebar--close"
>
  <div
    className="dashboard-sidebar__toggle"
  >
    <span
      className="dashboard-sidebar__close"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M689 165.1L308.2 493.5c-10.9 9.4-10.9 27.5 0 37L689 858.9c14.2 12.2 35 1.2 35-18.5V183.6c0-19.7-20.8-30.7-35-18.5z"
        />
      </svg>
    </span>
    <span
      className="dashboard-sidebar__open"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
        />
      </svg>
    </span>
  </div>
  <span
    className="dashboard-sidebar__logo"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
      >
        <img
          alt="euka logo"
          className="logo__image"
          src="test-file-stub"
        />
      </a>
    </div>
  </span>
  <ul
    className="dashboard-sidebar__list"
  >
    <a
      className="dashboard-sidebar__link"
      href="/"
    >
      <li
        className="dashboard-sidebar__row "
      >
        <span
          className="dashboard-sidebar__icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </span>
        <div
          className="dashboard-sidebar__title "
        >
          Home
        </div>
      </li>
    </a>
    <a
      className="dashboard-sidebar__link"
      href="/#analytics"
    >
      <li
        className="dashboard-sidebar__row "
      >
        <span
          className="dashboard-sidebar__icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 14h14v2h-16v-16h2zM4.5 13c-0.828 0-1.5-0.672-1.5-1.5s0.672-1.5 1.5-1.5c0.044 0 0.088 0.002 0.131 0.006l1.612-2.687c-0.154-0.235-0.243-0.517-0.243-0.819 0-0.828 0.672-1.5 1.5-1.5s1.5 0.672 1.5 1.5c0 0.302-0.090 0.583-0.243 0.819l1.612 2.687c0.043-0.004 0.087-0.006 0.131-0.006 0.033 0 0.066 0.001 0.099 0.004l2.662-4.658c-0.165-0.241-0.261-0.532-0.261-0.845 0-0.828 0.672-1.5 1.5-1.5s1.5 0.672 1.5 1.5c0 0.828-0.672 1.5-1.5 1.5-0.033 0-0.066-0.001-0.099-0.004l-2.662 4.658c0.165 0.241 0.261 0.532 0.261 0.845 0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5c0-0.302 0.090-0.583 0.243-0.819l-1.612-2.687c-0.043 0.004-0.087 0.006-0.131 0.006s-0.088-0.002-0.131-0.006l-1.612 2.687c0.154 0.235 0.243 0.517 0.243 0.819 0 0.828-0.672 1.5-1.5 1.5z"
            />
          </svg>
        </span>
        <div
          className="dashboard-sidebar__title "
        >
          Analytics
        </div>
      </li>
    </a>
    <a
      className="dashboard-sidebar__link"
      href="/#mailbox"
    >
      <li
        className="dashboard-sidebar__row "
      >
        <span
          className="dashboard-sidebar__icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </span>
        <div
          className="dashboard-sidebar__title "
        >
          Mailbox
        </div>
      </li>
    </a>
    <a
      className="dashboard-sidebar__link"
      href="/#family"
    >
      <li
        className="dashboard-sidebar__row "
      >
        <span
          className="dashboard-sidebar__icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
        </span>
        <div
          className="dashboard-sidebar__title "
        >
          Family
        </div>
      </li>
    </a>
  </ul>
</div>
`;
